### PR TITLE
[release-2.6] Validate query inputs

### PR DESCRIPTION
**Pull request by jlpadilla**
**Related Issue:**  https://github.com/stolostron/backlog/issues/23796

### Description of changes

- Merge #203 to release-2.6 branch
- Adds input validation.

### CODE REVIEW

File: src/v2/models/search.js [hunk 1]
```diff
@@ -14,10 +14,14 @@ import config from '../../../config';
...
 import { checkSearchServiceStatus } from './searchServiceStatus';
 
-// Remove single and double quotes because these can be used to inject malicious
-// code in the RedisGraph query. (SQL injection).
+// Validate that input only contains valid characters that won't allow injection
+// of malicious code in the query. (SQL injection)
+// Valid characters are a-z A-Z 0-9 - _ ! < > = . : /
 function sanitizeString(s) {
-  return s.replace(/[^a-zA-Z0-9\-_!<>=.:/]/g, ''); // Less risk of injection, but could be missing valid chars.
+  if (s.match(/[^a-zA-Z0-9\-_!<>=.:/]/g)) {
```
Comment (SherinV) [inline src/v2/models/search.js:21]: We have to do the same on V2.(?)

Comment (jlpadilla) [reply src/v2/models/search.js:21]: In v2, `goqu` should take care of the input sanitation, but we'll need to test to verify errors are reported correctly.

**Review by xrajesh — APPROVED** [review]

**Comment by xrajesh** [comment]
@atifshafi QE Required for this .

**Review by SherinV — APPROVED** [review]

**Pull request merged**